src/Hl7.Fhir.Support.Poco/Model/PrimitiveType.cs

```diff
@@ -21,17 +22,23 @@ namespace Hl7.Fhir.Model
     [DataContract]
     public abstract class PrimitiveType : DataType
     {
+        /// <inheritdoc />
         public override string TypeName { get { return "PrimitiveType"; } }
 
-        public object ObjectValue { get; set; }
+        /// <summary>
+        /// The value of the primitive, stored as an object.
+        /// </summary>
+        /// <remarks>This value maybe incompatible with the primitive's <c>Value</c> property. Both
+        /// <c>Value</c> and <c>ObjectValue</c> may contain invalid values according to the 
+        /// primitive's official domain. E.g. <c>Value</c> is a <c>string</c> for <see cref="FhirDateTime"/>,
+        /// and may contain illegally formatted values.</remarks>
+        public object? ObjectValue { get; set; }
```

Comment:
In 3.x and before, this behaviour was already there. *But* the old parser never really used this facility, only illegal values for enums were allowed. In all other cases the parser would throw an error if there was an illegal value.  It's even more subtle: in the new parsers, if there are no errors, we have practically the same behaviour (*if* we decide to add this validation), there will only be illegal values in the `PartialResult` of the deserialization exception otherwise.

Reply:
Decision: a *very* minor breaking change, will document the new situation where this property will *always* contain string when there is a failure.